samples/cors/arm-template.json

```diff
@@ -0,0 +1,159 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "baseName": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]",
+      "metadata": {
+        "description": "The base resource name."
+      }
+    }
+  },
+  "variables": {
+    "serviceName": "[concat(parameters('baseName'), '-api')]",
+    "vaultName": "[concat(parameters('baseName'), '-keyvault')]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.ApiManagement/service",
+      "apiVersion": "2019-01-01",
+      "name": "[variables('serviceName')]",
+      "location": "[resourceGroup().location]",
+      "properties": {
+        "notificationSenderEmail": "apimgmt-noreply@mail.windowsazure.com",
+        "enableClientCertificate": true,
+        "virtualNetworkType": "External",
+        "publisherEmail": "noreply@example.com",
+        "publisherName": "api-admin"
+      },
+      "sku": {
+        "name": "Developer"
+      },
+      "identity": {
+        "type": "SystemAssigned"
+      },
+      "resources": [
+        {
+          "type": "properties",
+          "apiVersion": "2019-01-01",
+          "name": "vault-name",
+          "dependsOn": [
+            "[resourceId('Microsoft.ApiManagement/service', variables('serviceName'))]"
+          ],
+          "properties": {
+            "displayName": "vault-name",
+            "value": "[variables('vaultName')]",
+            "secret": false
+          }
+        },
+        {
+          "type": "properties",
+          "apiVersion": "2019-01-01",
+          "name": "secret-name",
+          "dependsOn": [
+            "[resourceId('Microsoft.ApiManagement/service', variables('serviceName'))]"
+          ],
+          "properties": {
+            "displayName": "secret-name",
```

Comment:
Probably not a big deal but I'm wondering - if the user inputs a keyvault where they dno't actually have secret access does it just error out? Or fail silently?

Reply:
Requests will fail with an error that tries to provide context :) not silently!